akka-persistence/src/main/scala/akka/persistence/PersistencePlugin.scala

```diff
@@ -80,7 +80,12 @@ private[akka] abstract class PersistencePlugin[ScalaDsl, JavaDsl, T: ClassTag](s
 
     instantiate(
       (classOf[ExtendedActorSystem], system) :: (classOf[Config], pluginConfig) ::
-      (classOf[String], configPath) :: Nil)
+      (classOf[String], configPath) :: (classOf[Config], mergedConfig) :: Nil)
```

Comment:
Hmm, I'm missing something here, the `pluginConfig` is already the `mergedConfig.getConfig(configPath)` so why do we need to pass the `mergedConfig` as well?

Reply:
Thanks for your feedback @johanandren.

The `pluginConfig` is not the full config, but only a subset, more precisely, the read journal configuration.
Currently, in order to access the whole config, the extensions can use the actor system underlying configuration. Though, this actor system configuration might not contain the required information when using the mechanism put in place in https://github.com/akka/akka/issues/23618.
The `mergedConfig` above is that whole configuration in which the extensions can look for some information, and in the case of the JDBC implementation, it's where the plugin will find the `write-plugin` configuration. You could take a look at that change in my fork of `akka-persistence-jdbc` to get the big picture of that fix: https://github.com/akka/akka-persistence-jdbc/compare/v3.5.3...rdubois:v3.5.3-talend#diff-91148fd205afaadc748e53c2afaa35d1R78

Reply:
I think there's still something I don't get, is it to be able to override settings that are not in the config scope of the plugin, in what scenario do you need to do that?

Reply:
Our scenario is to allow to provide slick configuration at runtime for the persistence of our application. `akka-persistence` requires that the slick configuration is known when the actor system starts. It's not the case in our scenario. Some new configuration can be created on the fly. This was the purpose of #23618.
This being said, that means that the `write-plugin` of the JDBC extension can reference a config which is not part of the actor system config, but part of the runtime config. See the second argument of https://github.com/akka/akka/blob/master/akka-persistence/src/main/scala/akka/persistence/Persistence.scala#L300

Reply:
@johanandren if it's still not very clear, I can prepare an example which will highlight the issue.

Reply:
Thanks for the explanation, I think that was the piece of the puzzle I was missing.